ndcube/tests/test_ndcube.py

```diff
@@ -827,3 +827,133 @@ def test_reproject_exact_incompatible_wcs(ndcube_4d_ln_l_t_lt, wcs_4d_lt_t_l_ln,
+    u.Quantity(np.random.rand(10, 12), u.ct),
+])
+def test_map_arithmetic_rsubtract(ndcube_2d_ln_lt_units, value):
+    cube_quantity = u.Quantity(ndcube_2d_ln_lt_units.data, ndcube_2d_ln_lt_units.unit)
+    new_cube = value - ndcube_2d_ln_lt_units
+    check_arithmetic_value_and_units(new_cube, value - cube_quantity)
+
+
+@pytest.mark.parametrize('value', [
+    10 * u.ct,
+    u.Quantity([10], u.ct),
+    u.Quantity(np.random.rand(12), u.ct),
+    u.Quantity(np.random.rand(10, 12), u.ct),
+    10.0,
+    np.random.rand(12),
+    np.random.rand(10, 12),
+])
+def test_map_arithmetic_multiply(ndcube_2d_ln_lt_units, value):
+    cube_quantity = u.Quantity(ndcube_2d_ln_lt_units.data, ndcube_2d_ln_lt_units.unit)
+    new_cube = ndcube_2d_ln_lt_units * value
+    check_arithmetic_value_and_units(new_cube, cube_quantity * value)
+
+
+@pytest.mark.parametrize('value', [
+    10 * u.ct,
+    u.Quantity([10], u.ct),
+    u.Quantity(np.random.rand(12), u.ct),
+    u.Quantity(np.random.rand(10, 12), u.ct),
+    10.0,
+    np.random.rand(12),
+    np.random.rand(10, 12),
+])
+def test_map_arithmetic_rmultiply(ndcube_2d_ln_lt_units, value):
+    cube_quantity = u.Quantity(ndcube_2d_ln_lt_units.data, ndcube_2d_ln_lt_units.unit)
+    new_cube = value * ndcube_2d_ln_lt_units
+    check_arithmetic_value_and_units(new_cube, value * cube_quantity)
+
+
+@pytest.mark.parametrize('value', [
+    10 * u.ct,
+    u.Quantity([10], u.ct),
+    u.Quantity([2], u.s),
+    u.Quantity(np.random.rand(12), u.ct),
+    u.Quantity(np.random.rand(10, 12), u.ct),
+    10.0,
+    np.random.rand(12),
+    np.random.rand(10, 12),
+])
+def test_map_arithmetic_divide(ndcube_2d_ln_lt_units, value):
+    cube_quantity = u.Quantity(ndcube_2d_ln_lt_units.data, ndcube_2d_ln_lt_units.unit)
+    new_cube = ndcube_2d_ln_lt_units / value
+    check_arithmetic_value_and_units(new_cube, cube_quantity / value)
+
+
+@pytest.mark.parametrize('value', [
+    10 * u.ct,
+    u.Quantity([10], u.ct),
+    u.Quantity(np.random.rand(12), u.ct),
+    u.Quantity(np.random.rand(10, 12), u.ct),
+    10.0,
+    np.random.rand(12),
+    np.random.rand(10, 12),
+])
+def test_map_arithmetic_rdivide(ndcube_2d_ln_lt_units, value):
+    cube_quantity = u.Quantity(ndcube_2d_ln_lt_units.data, ndcube_2d_ln_lt_units.unit)
+    with pytest.warns(RuntimeWarning, match='divide by zero encountered in'):
+        new_cube = value / ndcube_2d_ln_lt_units
+        check_arithmetic_value_and_units(new_cube, value / cube_quantity)
+
+
+def test_map_arithmetic_pow(ndcube_2d_ln_lt_units):
+    cube_quantity = u.Quantity(ndcube_2d_ln_lt_units.data, ndcube_2d_ln_lt_units.unit)
+    new_cube = ndcube_2d_ln_lt_units ** 2
+    check_arithmetic_value_and_units(new_cube, cube_quantity ** 2)
+
+
+def test_map_arithmetic_neg(ndcube_2d_ln_lt_units):
+    check_arithmetic_value_and_units(
+        -ndcube_2d_ln_lt_units,
+        u.Quantity(-ndcube_2d_ln_lt_units.data, ndcube_2d_ln_lt_units.unit),
+    )
```

Comment:
Should include tests ensuring `NotImplementedError` is raised when it should be.

Reply:
Yes, good point. I will add those.